d2/src/main/java/com/linkedin/d2/balancer/util/ClusterInfoProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+   Copyright (c) 2019 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.d2.balancer.util;
+
+import com.linkedin.d2.balancer.ServiceUnavailableException;
+
+
+/**
+ * ClusterInfoProvider provides a mechanism to access detailed cluster information from the D2 infrastructure.
+ *
+ * @author David Hoa
+ * @version $Revision: $
+ */
+public interface ClusterInfoProvider
+{
+  String HTTPS_SCHEME = "https";
+  int DEFAULT_PARTITION = 0;
+
+  /**
+   * Obtain d2 cluster count
+   * @return int
+   */
+  int getClusterCount(String clusterName, String scheme, int partitionId) throws ServiceUnavailableException;
```

Comment:
what do you think about providing a fully async method? One idea is that I could provide the async as a base, and the blocking/sync as a convenience.